### PR TITLE
Image uploader script

### PR DESCRIPTION
**Pull request by jprochazk**
Closes https://github.com/rerun-io/rerun/issues/2132

### What

Adds `just upload <image>`, which uploads the image to google cloud using `gsutil`. The resulting URL contains the content hash plus the original file name, which ensures that it is unique.

```
$ just upload image.png
https://static.rerun.io/dd31922030a6bf7223d5e3728d8da5407f4d6b1a_image.png
```

The script itself contains instructions for how to setup `gsutil`. It's really just a link to Google's tutorial. Eventually it will be documented separately together with the rest of the new workflow.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2164

### CODE REVIEW

**Comment by jprochazk** [comment]
Example image uploaded via this script: https://static.rerun.io/dd31922030a6bf7223d5e3728d8da5407f4d6b1a_minimal.png

File: scripts/upload_image.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Upload an image.")
+    parser.add_argument("file", type=str, help="Path to the image.")
+    args = parser.parse_args()
+
+    object_name = content_hash(args.file) + f"_{os.path.basename(args.file)}"
+
+    subprocess.check_output(["gsutil", "cp", args.file, f"gs://rerun-static-img/{object_name}"])
```
Comment (jleibs) [inline scripts/upload_image.py:34]: If you didn't want to use sub-process we could consider doing this directly with the google storage APIs, though it's a bit more verbose...
```
from google.cloud import storage
gcs_client = storage.Client()
static_bucket = gcs_client.bucket("rerun-builds")
destination = static_bucket.blob(object_name)
destination.upload_from_file(buffer)
```

File: justfile [hunk 2]
```diff
@@ -139,3 +139,6 @@ download-design-tokens:
 # Update the results of `insta` snapshot regression tests
 update-insta-tests:
     cargo test; cargo insta review
+
+upload *ARGS:
+    python3 "scripts/upload_image.py" {{ARGS}}
```
Comment (emilk) [inline justfile:144]: please configure your editor to add newlines to end of files (and to trim trailing whitespace)!

**Comment by emilk** [comment]
It would be nice to run `pngcrush -rem allb -brute -reduce` on all pngs, but I guess that is something we can add later (and perhaps even do in batch on everything uploaded)

File: justfile (hunk 2 above)
Comment (jprochazk) [reply justfile:144]: I do have it setup to do that, but I also have a habit of doing save without formatting when I'm working on files that my editor will format differently than how they're currently formatted, like the workflow yaml files. Not the case here though! Will fix.

File: scripts/upload_image.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+
+"""
+Upload an image to Google Cloud.
+
+To use this:
+- Install the Google Cloud CLI (https://cloud.google.com/storage/docs/gsutil_install)
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply scripts/upload_image.py:8]: I changed the script to use the gcloud client library. If `gcloud` works, hopefully `gcloud init` should too, which should be enough to use this

[the remark this answers is not in the document]
Comment (emilk) [reply scripts/upload_image.py]: Sadly I still get the same error 😭

[the remark this answers is not in the document]
Comment (jprochazk) [reply scripts/upload_image.py]: How did you install it? I remember I using apt (https://cloud.google.com/storage/docs/gsutil_install#deb)

[the remark this answers is not in the document]
Comment (emilk) [reply scripts/upload_image.py]: I followed their instructions and downloaded https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-432.0.0-darwin-x86_64.tar.gz

[the remark this answers is not in the document]
Comment (jprochazk) [reply scripts/upload_image.py]: You could try installing via `brew install --cask google-cloud-sdk`, I tried it on my mac and it seems to work

[the remark this answers is not in the document]
Comment (emilk) [reply scripts/upload_image.py]: I tried the `brew` version now; same problem. I'll take a deeper look after the 0.6 release

File: scripts/upload_image.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    return h.hexdigest()
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Upload an image.")
+    parser.add_argument("path", type=str, help="Path to the image.")
+    args = parser.parse_args()
+
+    hash = content_hash(args.path)
+    object_name = f"{hash}_{args.path}"
```
Comment (emilk) [inline scripts/upload_image.py:37]: Looks like if I do `scripts/upload_image.py /Users/emilk/image.pg`, that will result in a really bad object_name. I think we should just use the filename here, not the whole path

**Review by Wumpf — APPROVED** [review]
works like a charm for me :)

**Pull request merged**